app.js

```diff
@@ -167,6 +169,16 @@ async function transfer(sender_id, recipient_id, amount) {
       error: validation.error.details[0].message,
     };
   }
+  const requestHash = hashArguments(sender_id, recipient_id, amount);
+
+  const redisCheckResult = await checkRedisHash(sender_id, requestHash);
+
+  if (!redisCheckResult.success) {
+    return {
+      success: false,
+      error: 'Duplicate transaction',
+    };
+  }
```

Comment:
@uncoooloj this is where I used the implementation for #3.

Reply:
Seen it, what if transactions are just a few milli seconds apart?

That's my fear. The system won't be done with this IO so both will pass.

What do you think?

Reply:
Yeah, Redis also has implementations for [transactions](https://www.npmjs.com/package/redis#optimistic-locks) and [optimistic locking](https://www.npmjs.com/package/redis#optimistic-locks). 

I'll update it in the final version.

Reply:
so quick question, what if there was a request throttling to one request per 10 ms for example? sounds like a lazy hack but that + a transaction has solves our problem no?